### ISSUE TITLE
ZeroDivisionError: float division by zero when use TopicRank

### ISSUE THREAD

**Issue by JiaxiangBU** [boudinfl/pke#142]
I read https://github.com/boudinfl/pke/issues/27 but don't find the solution I face the same error when I use the TopicRank model.

```python
extractor.load_document(input = 'tmp/text.txt', language = 'en')
```

```python
# keyphrase candidate selection, in the case of TopicRank: sequences of nouns
# and adjectives (i.e. `(Noun|Adj)*`)
extractor.candidate_selection()
```

```python
# candidate weighting, in the case of TopicRank: using a random walk algorithm
extractor.candidate_weighting()
# extractor.candidate_weighting(alpha=1.1, threshold=0.74, method='average')
```

    ---------------------------------------------------------------------------

    ZeroDivisionError                         Traceback (most recent call last)

    <ipython-input-45-ebda94891fc2> in <module>
          1 # candidate weighting, in the case of TopicRank: using a random walk algorithm
    ----> 2 extractor.candidate_weighting()
          3 # extractor.candidate_weighting(alpha=1.1, threshold=0.74, method='average')
    

    d:\install\miniconda\lib\site-packages\pke\unsupervised\graph_based\topicrank.py in candidate_weighting(self, threshold, method, heuristic)
        204 
        205         # build the topic graph
    --> 206         self.build_topic_graph()
        207 
        208         # compute the word scores using random walk
    

    d:\install\miniconda\lib\site-packages\pke\unsupervised\graph_based\topicrank.py in build_topic_graph(self)
        181                             if p_j < p_i:
        182                                 gap -= len(self.candidates[c_j].lexical_form) - 1
    --> 183                             self.graph[i][j]['weight'] += 1.0 / gap
        184 
        185     def candidate_weighting(self,
    

    ZeroDivisionError: float division by zero

I try the solution from https://github.com/boudinfl/pke/issues/27

> extractor = pke.unsupervised.MultipartiteRank(input_file='path/to/input.txt')
> extractor.read_document(format='raw')
> extractor.candidate_selection(n=5)
> extractor.candidate_weighting(alpha=1.1, threshold=0.74, method='average')
> keyphrases = extractor.get_n_best(n=100, stemming=False, redundancy_removal=True)

But maybe some arguments are depreciated.

I got this error.

```python
#https://github.com/boudinfl/pke/issues/27
extractor = pke.unsupervised.MultipartiteRank(input_file='tmp/text.txt')
extractor.read_document(format='raw')
extractor.candidate_selection(n=5)
extractor.candidate_weighting(alpha=1.1, threshold=0.74, method='average')
keyphrases = extractor.get_n_best(n=100, stemming=False, redundancy_removal=True)
```

    ---------------------------------------------------------------------------

    TypeError                                 Traceback (most recent call last)

    <ipython-input-41-bc1507fcaf55> in <module>
          1 #https://github.com/boudinfl/pke/issues/27
    ----> 2 extractor = pke.unsupervised.MultipartiteRank(input_file='tmp/text.txt')
          3 extractor.read_document(format='raw')
          4 extractor.candidate_selection(n=5)
          5 extractor.candidate_weighting(alpha=1.1, threshold=0.74, method='average')
    

    TypeError: __init__() got an unexpected keyword argument 'input_file'

Here is the content in the `text.txt`

```
In this section, we compute exactly the expected value of the AUC over all classifications
with a fixed number of errors and compare that to the error rate.
Different classifiers may have the same error rate but different AUC values. Indeed, for a
given classification threshold θ, an arbitrary reordering of the examples with outputs more
than θ clearly does not affect the error rate but leads to different AUC values. Similarly,
one may reorder the examples with output less than θ without changing the error rate.
Assume that the number of errors k is fixed. We wish to compute the average value of the
AUC over all classifications with k errors. Our model is based on the simple assumption
that all classifications or rankings with k errors are equiprobable. One could perhaps argue
that errors are not necessarily evenly distributed, e.g., examples with very high or very low
ranks are less likely to be errors, but we cannot justify such biases in general.
For a given classification, there may be x, 0 ≤ x ≤ k, false positive examples. Since the
number of errors is fixed, there are k − x false negative examples. Figure 3 shows the corresponding configuration. The two regions of examples with classification outputs above
and below the threshold are separated by a vertical line. For a given x, the computation of
the AUC, A, as given by Eq. (1) can be divided into the following three parts:
```

**Comment by ygorg** [comment]
Hello,
Maybe your problem is related to #77.
A typical loop for extracting keyphrases would be:
```python
for d in document:
    extractor = pke.unsupervised.TopicRank()
    extractor.load_document(d)
    extractor.candidate_selection()
    extractor.candidate_weighting()
    kps = extractor.get_n_best()
```

The `load_document` function does not reinitialize the candidates, so when loading multiple file on the same extractor, it tries to compute distance to candidates that do not exist in the current document (at that can have the same offset which raise this error).

And yes the parameters of the solution #27 are now deprecated. The load_document function now takes the `input_file`.

**Comment by JiaxiangBU** [comment]
@ygorg Thanks It works for me from your code snippet.

<details>
<summary>Output</summary>


```python
document = '''In this section, we compute exactly the expected value of the AUC over all classifications
with a fixed number of errors and compare that to the error rate.
Different classifiers may have the same error rate but different AUC values. Indeed, for a
given classification threshold θ, an arbitrary reordering of the examples with outputs more
than θ clearly does not affect the error rate but leads to different AUC values. Similarly,
one may reorder the examples with output less than θ without changing the error rate.
Assume that the number of errors k is fixed. We wish to compute the average value of the
AUC over all classifications with k errors. Our model is based on the simple assumption
that all classifications or rankings with k errors are equiprobable. One could perhaps argue
that errors are not necessarily evenly distributed, e.g., examples with very high or very low
ranks are less likely to be errors, but we cannot justify such biases in general.
For a given classification, there may be x, 0 ≤ x ≤ k, false positive examples. Since the
number of errors is fixed, there are k − x false negative examples. Figure 3 shows the cor�responding configuration. The two regions of examples with classification outputs above
and below the threshold are separated by a vertical line. For a given x, the computation of
the AUC, A, as given by Eq. (1) can be divided into the following three parts:'''
```


```python
extractor = pke.unsupervised.TopicRank()
extractor.load_document(document)
extractor.candidate_selection()
extractor.candidate_weighting()
kps = extractor.get_n_best()
```


```python
kps
```




    [('examples', 0.11687080459848757),
     ('errors', 0.11568378120424071),
     ('classifications', 0.09905192012365337),
     ('auc', 0.08157627771543798),
     ('fixed number', 0.05805590649412031),
     ('rankings', 0.053442868556763466),
     ('value', 0.0374103637927379),
     ('low', 0.031596931905306665),
     ('regions', 0.028883493588889152),
     ('high', 0.02876028894117715)]




</details>
